src/_stories/execute/utilities.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+from _stories.exceptions import StoryDefinitionError
+from _stories.returned import Failure
+from _stories.returned import Result
+from _stories.returned import Skip
+from _stories.returned import Success
+
+
+def validate_result_type(method, result_type, result):
+    if hasattr(method, "story_name"):
+        method_name = "{}.{}".format(method.story_name, method.__name__)
+    else:
+        try:
+            method_name = method.__qualname__
```

Comment:
`method.__name__` works fine for history. We could use it here as well.

Reply:
Wouldn't it be nice to know where exactly this happened?